src/main/java/com/google/api/codegen/transformer/py/PythonImportSectionTransformer.java

```diff
@@ -151,6 +183,101 @@ private ImportFileView generateApiImport(GapicMethodContext context) {
     return new ArrayList<>(appImports);
   }
 
+  private List<ImportFileView> generateTypesProtoImports(
+      Model model, GapicProductConfig productConfig, SurfaceNamer namer) {
+    ModelTypeTable typeTable = emptyTypeTable(productConfig);
+    Set<ImportFileView> imports = new TreeSet<>(importFileViewComparator());
+
+    // Save proto file import names to the type table for disambiguation.
+    Set<ProtoFile> protoFileDependencies = getProtofileDependencies(model);
+    populateTypeTable(protoFileDependencies, typeTable);
+
+    // Get disambiguated imports.
+    for (Map.Entry<String, TypeAlias> entry : typeTable.getImports().entrySet()) {
+      imports.add(generateAppImport(entry.getKey(), entry.getValue().getNickname()));
+    }
+    return ImmutableList.<ImportFileView>builder().addAll(imports).build();
+  }
+
+  private ModelTypeTable emptyTypeTable(GapicProductConfig productConfig) {
+    return new ModelTypeTable(
+        new PythonTypeTable(productConfig.getPackageName()),
+        new PythonModelTypeNameConverter(productConfig.getPackageName()));
+  }
+
+  private List<ImportFileView> generateTypesExternalImports() {
+    return ImmutableList.of(createImport("google.gax.utils.messages", "get_messages"));
+  }
+
+  private List<ImportFileView> generateTypesStandardImports() {
+    return ImmutableList.of(createImport("__future__", "absoulute_import"), createImport("sys"));
+  }
+
+  private void populateTypeTable(Set<ProtoFile> protoFileDependencies, ModelTypeTable typeTable) {
+    for (ProtoFile protoFile : protoFileDependencies) {
+      // For python, adding a single message from the proto file to the type table will populate
+      // the type table with the correct imports.
+      ImmutableList<MessageType> messages = protoFile.getMessages();
+      if (!messages.isEmpty()) {
+        typeTable.getAndSaveNicknameFor(TypeRef.of(messages.get(0)));
+      }
+    }
+  }
+
+  private Set<ProtoFile> getProtofileDependencies(Model model) {
```

Comment:
optional: it is probably significantly less efficient, but the simpler way to do this would be to subclass the api-compiler [Visitor](https://github.com/googleapis/api-compiler/blob/cc52101f81eae31ebdac647f9664f5b32d2228a1/src/main/java/com/google/api/tools/framework/model/Visitor.java). Here's [a more complicated example](https://github.com/googleapis/api-compiler/blob/4621e9bb63a3903b6b2c4639f6560c9284585a0f/src/main/java/com/google/api/tools/framework/processors/resolver/SymbolTableBuilder.java#L45) of doing this; in your case, the subclass would be trivial (it just needs to add ProtoFiles to a Set when it visits them).